build.gradle

```diff
@@ -195,6 +195,31 @@ def getProjectVersion() {
     return ver
 }
 
+project('serializers') {
+    dependencies {
+        compile project(':common')
+        compile project(':client')
+        compile group: 'org.apache.avro', name: 'avro', version: avroVersion
```

Comment:
Created issue https://github.com/pravega/schema-registry/issues/45 to tackle this later as it would require considerable refactoring at this time.

Reply:
The way i am thinking is as follows:
have format specific serializers implemented in format specific module.
So one for protobuf, one for json, one for avro. Each such module will have a factory with only serializers/deserializers for that format. 

Then we will have a consolidated one, which is this library and factory class.  This is because here we have serializers which are multi format. 

So this way users can take dependency on one common module (all type 1 apps that want to work for multiple formats can depend on this one module). 

So the user contract, which is the SerializerFactory, doesnt change. 

Only the serializer implementations would move internal to their respective modules and this base module would take dependence on those.

Reply:
If applications are supposed to use the specific modules, why do we need the base module? If I understand correctly, the base module is a level of indirection to the specific modules.

Reply:
it serves as two purposes:
1. level of indirection
2. allows for multi-format serializers and deserializers

Reply:
I'm fine with this direction, @tkaitchuck needs to resolve this thread if he is satisfied.